{% extends 'wrapper.html' %}

{% block title %}Documentation{% endblock %}

{% block body %}

    <link href='static/lib/highlight/highlight.css' rel='stylesheet'/>
    <script src='static/lib/highlight/highlight.js'></script>
    <script>hljs.initHighlightingOnLoad();</script>
    <script src='static/js/documentation.js'></script>

    <div class='page' id='documentation-page'>
        <div class='ribbon'>
            <div id='documentation-nav' class='container'>
                <h1>Documentation</h1>
                <ol>
                    <li><a href='documentation#overview'>Overview</a></li>
                    <li><a href='documentation#understanding-reports'>Understanding Reports</a></li>
                    <li><a href='documentation#custom-reports'>Custom Reports</a></li>
                    <li><a href='documentation#methods'>Methods</a></li>
                    <li><a href='documentation#api'>API</a></li>
                </ol>
            </div>
        </div>

        <div id='sections' class='container'>
            <div>
                <h2 id='overview'>Overview</h2>
                <div class='section'>
                    <p>To get started with GEN3VA, you first need to create a collection of tagged gene expression signatures
                        from GEO using <a href='http://amp.pharm.mssm.edu/g2e/' target='_blank'>GEO2Enrichr</a>. GEO2Enrichr is
                        a browser extension software tool that enables you to easily extract and tag signatures from GEO,
                        automatically adding these signatures to GEN3VA.</p>
                </div>
                <div class='section'>
                    <h4>Installing and using GEO2Enrichr</h4>
                    <p>To install GEO2Enrichr and to learn how to use it, visit <a href='http://amp.pharm.mssm.edu/g2e'
                                                                                   target='_blank'>GEO2Enrichr's website</a>.
                    </p>
                </div>
                <div class='section'>
                    <h4>Tags and reports</h4>
                    <p>A <em>tag</em> is a plain text term that you can use to associate multiple signatures from different
                        studies. These tags are non-hierarchical keywords assigned to a gene signature. In social media, the
                        equivalent idea is a 'hashtag', which is a tag preceded by a '#' or hash symbol. These tags enable the
                        creation of collections of gene signatures around common themes. A gene signature can have multiple
                        tags. Tagging is handled by GEO2Enrichr and is done when the signature is processed. See <a
                                href='http://amp.pharm.mssm.edu/g2e/manual#tags' target='_blank'>GEO2Enrichr's documentation</a>
                        for more details.</p>
                    <p>A <em>report</em> is a GEN3VA page with multiple visualizations and analyses of gene signatures under a
                        single tag. For example, the gene signatures associated with the tag
                        <em>AGING_BD2K_LINCS_DCIC_COURSERA</em> can be found <a
                                href='http://amp.pharm.mssm.edu/gen3va/tag/AGING_BD2K_LINCS_DCIC_COURSERA'
                                target='_blank'>here</a>, while the related report can be found <a
                                href='http://amp.pharm.mssm.edu/gen3va/report/approved/AGING_BD2K_LINCS_DCIC_COURSERA'
                                target='_blank'>here</a>. A report can also be generated from a subset of signatures associated
                        with a tag. This can be done using the <a href='documentation#custom-reports'>custom report builder</a> on each tag page.</p>
                </div>
            </div>

            <div>
                <h2 id='understanding-reports'>Understanding reports</h2>
                <div class='section'>
                    <p>Finding consensus across gene signatures profiled under the same or similar conditions by independent
                        studies can address important issues such as data reproducibility, assist in better understanding common
                        biological mechanisms, or facilitate drug discovery by identifying consensus drugs that can mimic or
                        reverse gene expression across a collection of signatures. The key idea underpinning GEN3VA is that gene
                        signatures under a common biological theme may share common properties, such as genes in their gene
                        lists, enrichment terms for gene set enrichment analysis results, or drugs that are predicted to reverse
                        or mimic expression. A <em>report</em> is best understood as aggregated information, a consolidation of
                        metadata, analyses, and visualizations on a collection of gene signatures around a common theme in an
                        attempt to find consensus between signatures.</p>
                </div>
                <div class='section'>
                    <h4>PCA plots and heatmaps</h4>
                    <div>
                        <p>GEN3VA reports provide principal component analysis (Fig. 1) and heatmaps (Fig. 2)
                            plots of the gene signatures in a collection. These plots are interactive; you can rotate, zoom, and
                            mouse over the plots to view more information. The heatmaps are built using <a
                                    href='http://amp.pharm.mssm.edu/clustergrammer/' target='_blank'>Clustergrammer</a>, a
                            visualization application developed by the Ma'ayan Lab.</p>
                        <img src='static/images/screenshots/screenshot_pca.png'/>
                        <p class='caption'>Fig. 1: Screenshot of the principal component analysis plot of 116 gene signatures extracted from studies that profiled the effect of adding endogenous ligands to mammalian cells.</p>
                    </div>
                    <div>
                        <h4>Manipulating the heatmaps</h4>
                        <p>
                            <a href='http://amp.pharm.mssm.edu/clustergrammer/' target='_blank'>Clustergrammer</a> provides a good video tutorial for interacting with the heatmaps, but GEN3VA has a couple additional features. First, GEN3VA heatmaps all have a <span class='mono'>dataset</span> category. Gene signatures from the same study will be grouped in the same category, the category being the study or dataset name. This is useful for checking if clusters in the heatmap are merely artifacts of duplicate data or are genuinely interesting.
                        </p>
                        <p>
                            Second, you can remove a column from a dataset using <span class='mono'>shift + click</span>. This allows you to probe what the heatmap will look like with specific signatures removed. A single click will sort the column.
                        </p>
                        <p>
                            These changes are not permanent, but you can build a custom report with any subset of signatures you would like using the <a href='documentation#custom-reports'>custom report builder</a>.
                        </p>
                        <img src='static/images/screenshots/screenshot_heatmap.png'/>
                        <p class='caption'>Fig. 2: Screenshot of the heatmap of the 116 gene signatures from
                            the endogenous ligand collection.</p>
                    </div>
                </div>
                <div class='section'>
                    <h4>Enrichment vector analysis with Enrichr</h4>
                    <p><a href='http://amp.pharm.mssm.edu/Enrichr/' target='_blank'>Enrichr</a> is a web application for
                        performing enrichment analysis on individual lists of genes. The input is a list of genes; optionally,
                        each gene can be followed by a weight indicating the degree of membership in the gene list. The output
                        is a list of enrichment terms from many gene set libraries. GEN3VA performs enrichment vector analysis
                        with Enrichr by submitting each gene signature from a tagged collection in a report and then clustering
                        the resultant terms.</p>
                </div>
                <div class='section'>
                    <h4>L1000CDS2 plots</h4>
                    <p><a href='http://amp.pharm.mssm.edu/L1000CDS2' target='_blank'>L1000CDS2</a> is a web application that
                        queries gene expression signatures against the LINCS L1000 data to identify and prioritize small
                        molecules that can either reverse or mimic the observed differential expression from the input
                        signature. GEN3VA creates L1000CDS2 heatmaps by performing this analysis for every
                        gene signature in the tagged collection and then clustering the scores for mimicking and reversing the
                        expression pattern from the input signatures.</p>
                </div>
            </div>

            <div>
                <h2 id='custom-reports'>Custom Reports</h2>
                <div class='section'>
                    <p>Sometimes a researcher may want to build a report from a subset of gene signatures from an existing collection. There are few a reasons to do this, such as wanting to categorize the signatures using a different metadata field or wanting to remove specific signatures for quality control reasons.</p>
                    <p>
                        GEN3VA has a custom report builder (Fig. 3) that allows a user to select all or some of the gene signatures from an existing collection and build a report. To build a custom report for a collection, click on the <strong>Signatures and Custom Report Builder</strong> button located at the top of any report page and follow the instructions there.</p>
                    <img src='static/images/screenshots/screenshot_custom_reports.png'/>
                        <p class='caption'>Fig. 3: Screenshot of a user building a custom report from the AGING_BD2K_LINCS_DCIC_COURSERA collection. Gene signature selection not shown.</p>
                </div>
            </div>

            <div>
                <h2 id='methods'>Methods</h2>
                <div class='section'>
                    <h4>Implementation details and database</h4>
                    <p>The GEN3VA web server is <a href='http://flask.pocoo.org/docs/0.10/' target='_blank'>Flask</a>
                        application running in an <a href='https://httpd.apache.org/' target='_blank'>Apache HTTP Server</a>
                        with the Apache WSGI module installed. Flask is a <a
                                href='https://en.wikipedia.org/wiki/Web_Server_Gateway_Interface'
                                target='_blank'>WSGI-compliant</a> framework for building web applications in Python. The
                        application and its dependencies are packaged and deployed in a Docker virtual machine onto a 16-node
                        computer cluster maintained by the Ma'ayan Lab.</p>
                    <p>The GEN3VA database runs on an internal <a href='https://mariadb.org/' target='_blank'>MariaDB</a>
                        server, a drop-in replacement for MySQL, that is maintained by the Ma'ayan Lab. This database is common
                        to both GEN3VA and GEO2Enrichr applications. Both applications use <a href='http://www.sqlalchemy.org/'
                                                                                              target='_blank'>SQLAlchemy ORM</a>.
                        An ORM (Object-Relational Mapping) is a framework that maps a tabular schema onto an object paradigm.
                        GEN3VA and GEO2Enrichr share ORM models and utility functions for accessing gene signatures through an
                        external project called <a href='https://github.com/MaayanLab/substrate' target='_blank'>Substrate</a>.

                    </p>
                </div>
            </div>

            <div>
                <h2 id='api'>API</h2>
                <div class='section'>
                    <h4>POST /gen3va/api/1.0/upload</h4>
                    <p>Uploads complete gene signature to GEN3VA's database. Does not perform any cleanup or analysis.</p>
                    <table class='table'>
                        <tr>
                            <td class='mono'>ranked_genes</td>
                            <td>A list of ranked genes represented as an array of arrays. For each inner array, the first element is the gene symbol and the second element is gene's weight or value.</td>
                        </tr>
                        <tr>
                            <td class='mono'>diffexp_method</td>
                            <td>Differential expression method. Default is <span class='mono'>chdir</span>.</td>
                        </tr>
                        <tr>
                            <td class='mono'>cutoff</td>
                            <td>Optional. The maximum number of genes in the resultant gene list. Defaults to 500.</td>
                        </tr>
                        <tr>
                            <td class='mono'>correction_method</td>
                            <td>Only applicable if <span class='mono'>diffexp_method</span> is <span class='mono'>ttest</span>. Defaults to <span class='mono'>BH</span> for Benjamini Hochberg.</td>
                        </tr>
                        <tr>
                            <td class='mono'>threshold</td>
                            <td>Only applicable if <span  class='mono'>diffexp_method</span> is <span class='mono'>ttest</span>. Defaults to 0.01.</td>
                        </tr>
                        <tr>
                            <td class='mono'>gene</td>
                            <td>Optional. Name or symbol for relevant gene.</td>
                        </tr>
                        <tr>
                            <td class='mono'>cell</td>
                            <td>Optional. Name of relevant cell type or tissue.</td>
                        </tr>
                        <tr>
                            <td class='mono'>perturbation</td>
                            <td>Optional. Name of relevant perturbation.</td>
                        </tr>
                        <tr>
                            <td class='mono'>disease</td>
                            <td>Optional. Name of relevant disease.</td>
                        </tr>
                        <tr>
                            <td class='mono'>tags</td>
                            <td>Optional. An array of tag names to be assigned to the gene signature.</td>
                        </tr>
                    </table>
                    <div>
                        <h4>Example using the <a href='http://docs.python-requests.org/en/master/' target='_blank'>requests</a> library.</h4>
                        <pre><code class='python hljs'>import json
import requests

payload = {
    'ranked_genes': [
        [
            'CPSF3',
            0.000631847
        ],
        [
            'CLEC18B',
            0.00876892
        ],
        [
            'RTDR1',
            0.0000692485
        ],
        [
            'MYLPF',
            0.00218427
        ],
        [
            'KIF2B',
            0.0000457653
        ],
        [
            'SMPDL3A',
            0.00876879
        ],
        [
            'FAM171A2',
            0.00442025
        ],
        [
            'RASGRF2',
            0.0145588
        ],
        [
            'ROCK2',
            0.00218436
        ]
    ],
    'diffexp_method': 'chdir',
    'tags': ['test_tag'],
    'gene': 'STAT3',
    'cell': None,
    'perturbation': None,
    'disease': None
}
resp = requests.post('http://amp.pharm.mssm.edu/gen3va/api/1.0/upload',
                     data=json.dumps(payload))</code></pre>
                    </div>
                </div>
            </div>
        </div>
    </div>

{% endblock %}