{% extends 'wrapper.html' %}

{% block title %}Documentation{% endblock %}

{% block body %}

    <div class="page" id="documentation-page">
        <h1>Getting Started</h1>

        <div class="section">
            <p>To get started with GEN3VA, you first need to create a collection of tagged gene expression signatures
                from GEO using <a href="http://amp.pharm.mssm.edu/g2e/" target="_blank">GEO2Enrichr</a>. GEO2Enrichr is
                a browser extension software tool that enables you to easily extract and tag signatures from GEO,
                automatically adding these signatures to GEN3VA.</p>
        </div>

        <div class="section">
            <h4>Installing and using GEO2Enrichr</h4>
            <p>To install GEO2Enrichr and to learn how to use it, visit <a href="http://amp.pharm.mssm.edu/g2e"
                                                                           target="_blank">GEO2Enrichr's website</a>.
            </p>
        </div>

        <div class="section">
            <h4>Tags and reports</h4>
            <p>A <em>tag</em> is a plain text term that you can use to associate multiple signatures from different
                studies. These tags are non-hierarchical keywords assigned to a gene signature. In social media, the
                equivalent idea is a "hashtag", which is a tag preceded by a "#" or hash symbol. These tags enable the
                creation of collections of gene signatures around common themes. A gene signature can have multiple
                tags. Tagging is handled by GEO2Enrichr and is done when the signature is processed. See <a
                        href="http://amp.pharm.mssm.edu/g2e/manual#tags" target="_blank">GEO2Enrichr's documentation</a>
                for more details.</p>
            <p>A <em>report</em> is a GEN3VA page with multiple visualizations and analyses of gene signatures under a
                single tag. For example, the gene signatures associated with the tag
                <em>AGING_BD2K_LINCS_DCIC_COURSERA</em> can be found <a
                        href="http://amp.pharm.mssm.edu/gen3va/tag/AGING_BD2K_LINCS_DCIC_COURSERA"
                        target="_blank">here</a>, while the related report can be found <a
                        href="http://amp.pharm.mssm.edu/gen3va/report/approved/AGING_BD2K_LINCS_DCIC_COURSERA"
                        target="_blank">here</a>. A report can also be generated from a subset of signatures associated
                with a tag. This can be done using the <em>Custom Report Builder</em> on each tag page.</p>
        </div>

        <h1>Understanding reports</h1>
        <div class="section">

            <p>Finding consensus across gene signatures profiled under the same or similar conditions by independent
                studies can address important issues such as data reproducibility, assist in better understanding common
                biological mechanisms, or facilitate drug discovery by identifying consensus drugs that can mimic or
                reverse gene expression across a collection of signatures. The key idea underpinning GEN3VA is that gene
                signatures under a common biological theme may share common properties, such as genes in their gene
                lists, enrichment terms for gene set enrichment analysis results, or drugs that are predicted to reverse
                or mimic expression. A <em>report</em> is best understood as aggregated information, a consolidation of
                metadata, analyses, and visualizations on a collection of gene signatures around a common theme in an
                attempt to find consensus between signatures.</p>
        </div>

        <div class="section">
            <h4>PCA and hierarchical clustering plots</h4>
            <div>
                <p>GEN3VA reports provide principal component analysis (Fig. 1) and hierarchical clustering (Fig. 2)
                    plots of the gene signatures in a collection. These plots are interactive; you can rotate, zoom, and
                    mouse over the plots to view more information. The hierarchical clustering plots are built using <a
                            href="http://amp.pharm.mssm.edu/clustergrammer/" target="_blank">Clustergrammer</a>, a
                    visualization application developed by the Ma'ayan Lab.</p>
                <img src="static/images/screenshots/screenshot_pca.png"/>
                <p class="caption">Fig. 1: Screenshot of the principal component analysis plot of 116 gene signatures
                    extracted from studies that profiled the effect of adding endogenous ligands to mammalian cells.</p>
                <img src="static/images/screenshots/screenshot_hierarchical_clustering.png"/>
                <p class="caption">Fig. 2: Screenshot of the hierarchical clustering of the 116 gene signatures from
                    the endogenous ligand collection.</p>
            </div>
        </div>

        <div class="section">
            <h4>Enrichment vector analysis with Enrichr</h4>
            <p><a href="http://amp.pharm.mssm.edu/Enrichr/" target="_blank">Enrichr</a> is a web application for
                performing enrichment analysis on individual lists of genes. The input is a list of genes; optionally,
                each gene can be followed by a weight indicating the degree of membership in the gene list. The output
                is a list of enrichment terms from many gene set libraries. GEN3VA performs enrichment vector analysis
                with Enrichr by submitting each gene signature from a tagged collection in a report and then clustering
                the resultant terms.</p>
        </div>

        <div class="section">
            <h4>L1000CDS2 plots</h4>
            <p><a href="http://amp.pharm.mssm.edu/L1000CDS2" target="_blank">L1000CDS2</a> is a web application that
                queries gene expression signatures against the LINCS L1000 data to identify and prioritize small
                molecules that can either reverse or mimic the observed differential expression from the input
                signature. GEN3VA creates L1000CDS2 hierarchical clustering plots by performing this analysis for every
                gene signature in the tagged collection and then clustering the scores for mimicking and reversing the
                expression pattern from the input signatures.</p>
        </div>

        <h1>Methods</h1>
        <div class="section">
            <h4>Implementation details and database</h4>
            <p>The GEN3VA web server is <a href="http://flask.pocoo.org/docs/0.10/" target="_blank">Flask</a>
                application running in an <a href="https://httpd.apache.org/" target="_blank">Apache HTTP Server</a>
                with the Apache WSGI module installed. Flask is a <a
                        href="https://en.wikipedia.org/wiki/Web_Server_Gateway_Interface"
                        target="_blank">WSGI-compliant</a> framework for building web applications in Python. The
                application and its dependencies are packaged and deployed in a Docker virtual machine onto a 16-node
                computer cluster maintained by the Ma'ayan Lab.</p>
            <p>The GEN3VA database runs on an internal <a href="https://mariadb.org/" target="_blank">MariaDB</a>
                server, a drop-in replacement for MySQL, that is maintained by the Ma'ayan Lab. This database is common
                to both GEN3VA and GEO2Enrichr applications. Both applications use <a href="http://www.sqlalchemy.org/"
                                                                                      target="_blank">SQLAlchemy ORM</a>.
                An ORM (Object-Relational Mapping) is a framework that maps a tabular schema onto an object paradigm.
                GEN3VA and GEO2Enrichr share ORM models and utility functions for accessing gene signatures through an
                external project called <a href="https://github.com/MaayanLab/substrate" target="_blank">Substrate</a>.

            </p>
        </div>
    </div>

{% endblock %}